R/makeTreeSummarizedExperimentFromqiime2.R

```diff
@@ -201,12 +219,14 @@ subset_taxa_in_feature <- function(taxa_tab, feature_tab) {
 #'  between different taxonomic levels, defaults to on compatible with both
 #'  GreenGenes and SILVA `; |;"`.
 #' @param trim_rank_prefix logical whether remove leading characters from
-#'   taxonomic levels, e.g. k__ or D_0__, default `TRUE`.
+#'   taxonomic levels, e.g. k__ or D_0__, default `FALSE`.
 #' @return [`phyloseq::taxonomyTable-class`] object.
 #' @keywords internal
 #' @importFrom rlang .data
+#' @importFrom tidyr separate
 #' @noRd
-parse_q2taxonomy <- function(taxa, sep = "; |;", trim_rank_prefix = TRUE) {
+.parse_q2taxonomy <- function(taxa, sep = "; |;", trim_rank_prefix = FALSE) {
+    stopifnot(is.logical(trim_rank_prefix))
```

Comment:
The test is totally fine, but I would like you to reconsider testing this very early on at the beginning of `makeTreeSummarizedExperimentFromqiime2`. The reasoning behind this, is that if a larger data set gets imported, an error might get thrown after most of the import work is already done.

A lot of people probably have said it:
> fail early, fail often